partner_multi_company/models/res_users.py

```diff
@@ -9,10 +9,13 @@ class ResUsers(models.Model):
 
     @api.model_create_multi
     def create(self, vals):
-        res = super(ResUsers, self).create(vals)
-        if "company_ids" in vals:
-            res.partner_id.company_ids = vals["company_ids"]
-        return res
+        users = super(ResUsers, self).create(vals)
```

Comment:
We're stumbling upon some cases where the constraint is raising before being able to ensure the compatibility post actions. So maybe it's useless to perform that check before the following code...

This context would be catched in the constraint, of course :)
```suggestion
        users = super(ResUsers, self.with_context(check_company_id=True)).create(vals)
```

Reply:
I'm not a fan of this approach because I've seen how activating/deactivating features with context can end up in a _flags mess_, but I agree it can be a solution, thanks for proposing it!
I would leave it as a last resort when everything else fails.

As for this particular case I have found how to make the tests happy, and I added a test with the use-case that was failing previously, please check!